packages/api/src/beacon/server/validator.ts

```diff
@@ -4,6 +4,24 @@ import {ServerRoutes, getGenericJsonServer} from "../../utils/server/index.js";
 import {ServerApi} from "../../interfaces.js";
 
 export function getRoutes(config: ChainForkConfig, api: ServerApi<Api>): ServerRoutes<Api, ReqTypes> {
-  // All routes return JSON, use a server auto-generator
-  return getGenericJsonServer<ServerApi<Api>, ReqTypes>({routesData, getReturnTypes, getReqSerializers}, config, api);
+  const reqSerializers = getReqSerializers();
+  const returnTypes = getReturnTypes();
+
+  // Most of routes return JSON, use a server auto-generator
+  const serverRoutes = getGenericJsonServer<ServerApi<Api>, ReqTypes>(
+    {routesData, getReturnTypes, getReqSerializers},
+    config,
+    api
+  );
+  return {
+    ...serverRoutes,
+    produceBlockV3: {
+      ...serverRoutes.produceBlockV3,
+      handler: async (req, res) => {
+        const response = await api.produceBlockV3(...reqSerializers.produceBlockV3.parseReq(req));
+        void res.header("Eth-Consensus-Version", response.version);
```

Comment:
Do we want set other headers here as well? My guess is that most should get those from the json body anyways though

Reply:
> Do we want set other headers here as well? My guess is that most should get those from the json body anyways though

I was trying to ask sproul if he needs the other headers but it doesn't seem like he needs them.

I guess it's easy enough to add the other headers. It doesn't hurt to add them now

Reply:
One optimization could be that you just look at the block value header and not even parse the body, might be what vouch is doing if it requests blocks from multiple bns